src/core/validation.py

```
@@ -455,10 +455,10 @@ def validate_crawl_dir_params(
     directory_path: str,
     max_files: int = 500,
     max_file_size: int = 1048576,
-    exclude_patterns: Optional[List[str]] = None,
-    include_patterns: Optional[List[str]] = None,
+    exclude_patterns: Optional[list[str]] = None,
```

Ruff (0.12.2)

Use `X | None` for type annotations

Convert to `X | None`

(UP045)